MaxText/layers/llama2.py

```diff
@@ -0,0 +1,159 @@
+"""
+ Copyright 2023 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ """
+
+"""Transformer model definition."""
+# pylint: disable=arguments-differ
+# pylint: disable=no-name-in-module
+
+from jax.sharding import Mesh
+
+
+
+from flax import linen as nn
+
+
+import jax.numpy as jnp
+# from jax.experimental.pallas.ops.tpu import flash_attention
+from layers import attentions
+from layers import embeddings
+from layers import linears
+from layers import normalizations
+
+from layers import models
+
+import common_types
+
+Array = common_types.Array
+Config = common_types.Config
+DType = common_types.DType
+Mesh = common_types.Mesh
+ScanIn = common_types.ScanIn
+
+Embed = embeddings.Embed
+LLaMARotaryEmbedding = embeddings.LLaMARotaryEmbedding
+Attention = attentions.Attention
+AttentionOp = attentions.AttentionOp
+RMSNorm = normalizations.RMSNorm
+
+
+#-----------------------------------------
+# The Decoder Layer specific for Llama2
+#-----------------------------------------
+
+
+class DecoderLayer(models.DecoderLayer):
```

Comment:
Do you really want to subclass here? (Are you using that subclassing?) I think it would be best not to encourage subclassing because we're saying there isn't reuse. If there is reuse... let's talk about it!

Reply:
Please name your class LlamaDecoderLayer

Reply:
Done - thanks!